File: lt1089.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    void duplicateZeros(vector<int>& arr) {
        int len = arr.size();
        vector<int> pos;
        for(int i = 0; i < len; i++)
            if(!arr[i])
                pos.push_back(i);
        for(int i = pos.size() - 1; i >=0;i++)
            arr.insert(arr.begin() + pos[i], 0);
        
        arr = vector<int>(arr.begin(), arr.begin() + len);

    }
};

File: lt1089.java
// https://leetcode-cn.com/problems/duplicate-zeros/
// duplicate zeros
class Solution {
    public void duplicateZeros(int[] arr) {
        List<Integer> res= new ArrayList<>();
        for(int e:arr){
            res.add(e);
            if(e==0)
                res.add(0);
        }
        for(int i=0;i<arr.length;i++){
            arr[i]=res.get(i);
        }
    }
}